### PR TITLE
feat(rtl): use gclient (gsdk) as remoting runtime, implement Remoting trait for gclient::GearApi

### LINKED ISSUE #296: feat(rtl): use gclient (gsdk) as remoting runtime, implement Remoting trait for gclient::GearApi
### Problem to Solve

Currently it is not possible to use generated client code from the gclient (gsdk)

### Possible Solution

Implement Remoting trait for gclient::GearApi

### Notes

_No response_

### PR DESCRIPTION
**Pull request by vobradovich**
Resolves #296 .

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
 env:
   BINARYEN_VERSION: version_111
   CARGO_TERM_COLOR: always
+  GEAR_PATH: /usr/bin/gear
```
Comment (DennisInSky) [inline .github/workflows/ci.yml:11]: Doesn't seem used

File: rtl/src/errors.rs [hunk 2]
```diff
@@ -18,6 +18,9 @@ pub enum Error {
     GCore(#[from] GCoreError),
     #[error("codec: {0}")]
     Codec(#[from] CodecError),
+    #[cfg(not(target_arch = "wasm32"))]
```
Comment (DennisInSky) [inline rtl/src/errors.rs:21]: Can we go w/o conditional compilation here?

File: rtl/src/errors.rs [hunk 3]
```diff
@@ -18,6 +18,9 @@ pub enum Error {
...
+    #[error("codec: {0}")]
+    GClient(#[from] gclient::Error),
```
Comment (DennisInSky) [inline rtl/src/errors.rs:23]: -> GSdk?

**Review by DennisInSky — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (vobradovich) [reply .github/workflows/ci.yml:11]: it used in test to run node
```
    let api = GearApi::dev_from_path(env!("GEAR_PATH")).await.unwrap();
```

File: rtl/src/errors.rs (hunk 3 above)
Comment (vobradovich) [reply rtl/src/errors.rs:23]: ` gclient::Error` is more obvious, and I don't like aliases)

File: rtl/src/errors.rs (hunk 2 above)
Comment (vobradovich) [reply rtl/src/errors.rs:21]: No, gclient does not compile for wasm32

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -79,3 +79,14 @@ jobs:
...
+      - name: Download Gear node
+        run: |
+          sudo wget -O ./gear https://github.com/gear-tech/gear/releases/download/build/gear
+          sudo chmod +x gear
+          sudo mv gear /usr/bin/
+
+      - name: Run Integration Tests on Gear Node
+        env:
+          GEAR_PATH: /usr/bin/gear
+        run: cargo test gclient -p demo --locked --no-fail-fast -- --ignored
```
Comment (DennisInSky) [inline .github/workflows/ci.yml:92]:
```suggestion
        run: cargo test -p demo --test counter_gclient --locked --no-fail-fast -- --ignored
```
Provided you rename `gclient_counter.rs` into `counter_gclient.rs`

File: examples/demo/wasm/tests/gclient_counter.rs
Comment (DennisInSky) [inline examples/demo/wasm/tests/gclient_counter.rs:1]: -> counter_gclient.rs
Probably it's worth renaming `counter.rs` into `counter_gtest.rs` as well

File: rtl/src/gsdk/calls.rs [hunk 5]
```diff
@@ -0,0 +1,113 @@
+use crate::{
+    calls::Remoting,
+    errors::{Result, RtlError},
+    ActorId, CodeId, GasUnit, ValueUnit, Vec,
+};
+use core::future::Future;
+use gclient::{ext::sp_core::ByteArray, EventProcessor, GearApi};
+use gprimitives::H256;
```
Comment (DennisInSky) [inline rtl/src/gsdk/calls.rs:8]: You could've used it from crate

**Review by DennisInSky — APPROVED** [review]

**Review by DennisInSky — APPROVED** [review]

**Pull request merged**